prow/gerrit/client/syncer_test.go

```diff
@@ -77,34 +77,33 @@ func TestSyncTime(t *testing.T) {
 	earlier := now.Add(-time.Hour)
 	later := now.Add(time.Hour)
 
-	if err := st.Update(LastSyncState{"foo": {"bar": earlier}}); err != nil {
+	if err := st.Update(LastSyncState{"foo": {"bar": earlier}}, true); err != nil {
 		t.Fatalf("Failed update: %v", err)
 	}
 	if actual := st.Current()["foo"]["bar"]; !actual.Equal(cur) {
 		t.Errorf("Update(%v) should not have reduced value from %v, got %v", earlier, cur, actual)
 	}
 
-	if err := st.Update(LastSyncState{"foo": {"bar": later}}); err != nil {
+	if err := st.Update(LastSyncState{"foo": {"bar": later}}, true); err != nil {
 		t.Fatalf("Failed update: %v", err)
 	}
 	if actual := st.Current()["foo"]["bar"]; !actual.After(cur) {
 		t.Errorf("Update(%v) did not move current value to after %v, got %v", later, cur, actual)
 	}
 
 	expected := later
-	st = SyncTime{
-		path:   path,
-		opener: open,
-		ctx:    ctx,
-	}
+	st = NewSyncTime(path, open, ctx)
+	st.uploadInterval = time.Microsecond * 10
+
+	time.Sleep(time.Second * 2)
```

Comment:
Nit: Somehow poll a channel to figure out how long to block (sort of like the pattern in https://github.com/kubernetes/test-infra/blob/c481b28dc87ec7c74c0513880d6a9faf56d616e6/prow/test/integration/test/pod-utils_test.go#L575) instead of sleeping.

Reply:
changed the implementation to wait for completion when `force` is true, so this is not necessary any more